### PR TITLE
Improvements for backup UI 

### PR DESCRIPTION
**Pull request by mikeger**
- Fixed automation.
- Added possibility to inject the database.
- Updated No History screen design.
- Fixed issue with adding email and going to back up again.
- Fixed loading overlay issue.

### CODE REVIEW

File: Wire-iOS/Sources/Components/Settings.m [hunk 1]
```diff
@@ -69,7 +69,6 @@ @interface Settings ()
 @property (strong, readonly, nonatomic) NSUserDefaults *defaults;
 @property (nonatomic) BOOL shouldSend500Messages;
 @property (nonatomic) NSTimeInterval maxRecordingDurationDebug;
-@property (nonatomic) ZMEmailCredentials *automationTestEmailCredentials;
```
Comment (mikeger) [inline Wire-iOS/Sources/Components/Settings.m]: This was not used anywhere, just caught my eye.

**Comment by daehn** [comment]
Might be good to set `snapshotBackgroundColor` to something non-clear in the snapshot tests, then it's easier to see what's going on.

**Comment by mikeger** [comment]
@daehn done

**Comment by mikeger** [comment]
Tested and it actually restores.

File: Wire-iOS/Sources/UserInterface/Registration/RegistrationPersonal/Steps/NoHistoryViewController.swift [hunk 2]
```diff
@@ -55,7 +82,16 @@ extension NoHistoryViewController {
     }
     
     fileprivate func showFilePicker() {
-        let picker = UIDocumentMenuViewController(documentTypes: [NoHistoryViewController.WireBackupUTI],
+        // Test code to verify restore
+        #if arch(i386) || arch(x86_64)
+            let testFilePath = "/var/tmp/backup.ios_wbu"
```
Comment (mikeger) [inline Wire-iOS/Sources/UserInterface/Registration/RegistrationPersonal/Steps/NoHistoryViewController.swift]: @mykola-mokhnach this is the path where the iOS app would expect the backup to be, similar to file sharing.

**Pull request merged**